### ISSUE TITLE
Fix `GetStackAllocConversion` and remove `Binder.CheckFeatureAvailability` method

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#25028]
During overload resolution or conversion classification, we use use-site diagnostics to report problems with symbols that might affect the correctness of result. 
`GetStackAllocConversion` should not be putting language-version-related diagnostics as use-site diagnostics since there was nothing wrong with symbols. Also, we don't want language version to affect the result of the binding (including conversion classification).

We should review this method's use of use-site diagnostics to comply with this guideline.

This was reported by @AlekseyTs in a PR that was also going to mis-use use-site diagnostics (https://github.com/dotnet/roslyn/pull/24913#discussion_r170335206).

```

        public override Conversion GetStackAllocConversion(BoundStackAllocArrayCreation sourceExpression, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
        {
...
                            Binder.CheckFeatureAvailability(sourceExpression.Syntax, MessageID.IDS_FeatureRefStructs, ref useSiteDiagnostics);
                            Binder.GetWellKnownTypeMember(_binder.Compilation, WellKnownMember.System_Span_T__ctor, out DiagnosticInfo memberDiagnosticInfo);
                            HashSetExtensions.InitializeAndAdd(ref useSiteDiagnostics, memberDiagnosticInfo);

                            return Conversion.MakeStackAllocToSpanType(spanConversion);
                        }
...
        }
```

Tagging @VSadov @OmarTawfik

**Comment by OmarTawfik** [comment]
I'll take this one.
